identity/client/src/pages/groups/detail/mapping-rules/AssignMappingRulesModal.tsx

```diff
@@ -37,25 +37,34 @@ const AssignMappingRulesModal: FC<
   const [loadingAssignMappingRule, setLoadingAssignMappingRule] =
     useState(false);
 
+  const [mappingRuleFilter, setMappingRuleFilter] = useState({});
+  const handleMappingRuleFilterChange = useCallback((search: string) => {
+    if (!search) {
```

Comment:
🔧 This applies to all other files with similar logic as well.
```suggestion
    if (!search.trim()) {
```